File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/unpacker/ChannelPostUnpackerParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.receiver.ChannelPostReceiver;
import ru.ioleksiv.telegram.bot.api.model.objects.std.Message;
import ru.ioleksiv.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class ChannelPostUnpackerParser extends UnpackerParser<ChannelPostReceiver, Message> {

    @Override
    public Class<ChannelPostReceiver> getAnnotationClass() {
        return ChannelPostReceiver.class;
    }

    @Override
    public UpdateUnpacker<Message> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getChannelPost()), Message.class);
    }

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/core/controller/annotations/parser/unpacker/ChannelPostUnpackerParser.kt
package io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.unpacker

import io.github.oleksivio.tl.kbot.core.annotations.receiver.ChannelPostReceiver
import io.github.oleksivio.tl.kbot.core.core.controller.handler.unpack.UpdateUnpacker
import io.github.oleksivio.tl.kbot.server.api.objects.std.Message

class ChannelPostUnpackerParser : UnpackerParser<ChannelPostReceiver, Message>() {
    override val updateUnpacker: UpdateUnpacker<Message>
        get() = UpdateUnpacker(
            { it.channelPost },
            Message::class
        )

    override val parserAnnotationClass = ChannelPostReceiver::class
}



